src/lib/ggshield-api.ts

```diff
@@ -172,47 +172,42 @@ export async function scanFile(
     filePath,
   ]);
 
-  // Ignore errors concerning usage
-  // This occurs when the path of the file is invalid (i.e.VSCode sending an event for files not on the file system)
-  // or when the file is ignored in the .gitguardian.yaml
-  if (
-    proc.stderr.includes(
-      "Error: An ignored file or directory cannot be scanned"
-    )
-  ) {
-    updateStatusBarItem(StatusBarStatus.ignoredFile);
-    return;
-  }
-  if (proc.stderr.includes("Usage: ggshield secret scan path")) {
-    return undefined;
-  }
-  let errorMessage = "";
-  proc.stderr.split("\n").forEach((stderrLine) => {
+  if (proc.status === 128 || proc.status === 3) {
+    let errorMessage = "";
+    proc.stderr.split("\n").forEach((stderrLine) => {
+      if (
+        stderrLine.length > 0 &&
+        !stderrLine.includes("Scanning Path...") // ggshield outputs this info message on stderr, ignore it
+      ) {
+        errorMessage += stderrLine + "\n";
+      }
+    });
+    if (errorMessage.length > 0) {
+      window.showErrorMessage(`ggshield: ${errorMessage}`);
+      return undefined;
+    }
+  } else if (proc.status === 2) {
+    // Ignore errors concerning usage
+    // This occurs when the path of the file is invalid (i.e.VSCode sending an event for files not on the file system)
+    // or when the file is ignored in the .gitguardian.yaml
     if (
-      stderrLine.length > 0 &&
-      !stderrLine.includes("Scanning Path...") // ggshield outputs this info message on stderr, ignore it
+      proc.stderr.includes(
+        "Error: An ignored file or directory cannot be scanned"
+      )
     ) {
-      errorMessage += stderrLine + "\n";
+      updateStatusBarItem(StatusBarStatus.ignoredFile);
+      return;
     }
-  });
-  if (errorMessage.length > 0) {
-    window.showErrorMessage(`ggshield: ${errorMessage}`);
     return undefined;
-  }
-
-  const results = JSON.parse(proc.stdout);
-  if (!results) {
-    updateStatusBarItem(StatusBarStatus.ready);
+  } else if (proc.status === 0) {
+    updateStatusBarItem(StatusBarStatus.noSecretFound);
     return;
```

Comment:
Here we still need to do `diagnosticCollection.set(fileUri, incidentsDiagnostics);`, or else the problem won't be removed when the secret is removed from the file.

I think that parseGGShieldResults also works when there are no results, so we can maybe just leave `updateStatusBarItem` in the if else and put everything else after ?

Reply:
Oh right :sweat_smile:  I might have been a bit violent while trying to remove old unnecessary code...